### PR TITLE
[T4A3][W15-B4]Jasper Lee

### PR DESCRIPTION
**Pull request by jasperlee27**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/data/person/Contact.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ 
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Address // instanceof handles nulls
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:24]: ouch! you see that you are comparing parent class with one other child class instance? think how you would alleviate this problem?

File: src/seedu/addressbook/data/person/Contact.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package seedu.addressbook.data.person;
+
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:2]: Please include a header comment to indicate what this class/method does?

File: src/seedu/addressbook/data/person/Email.java [hunk 3]
```diff
@@ -3,31 +3,27 @@
...
      *
-     * @throws IllegalValueException if given email address string is invalid.
+     * @throws IllegalValueException
+     *             if given email address string is invalid.
      */
     public Email(String email, boolean isPrivate) throws IllegalValueException {
-        this.isPrivate = isPrivate;
-        email = email.trim();
+        super(email.trim(), isPrivate);
+
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Email.java:23]: think how you can also push the validation logic to the parent class using the super constructor. :-)

**Comment by okkhoy** [comment]
@jasperlee27 
Some comments added. Please ack & close the PR after reading comments.

**Comment by jasperlee27** [comment]
Acknowledged. Thankyou!

**Pull request closed without merging**